diff --git a/cmake/install-rules.cmake b/cmake/install-rules.cmake
index 8b1914a..8442e8c 100644
--- a/cmake/install-rules.cmake
+++ b/cmake/install-rules.cmake
@@ -1,6 +1,6 @@
 if(PROJECT_IS_TOP_LEVEL)
   set(
-      CMAKE_INSTALL_INCLUDEDIR "include/zxorm-${PROJECT_VERSION}"
+      CMAKE_INSTALL_INCLUDEDIR "include"
       CACHE PATH ""
   )
 endif()
diff --git a/include/zxorm/orm/connection.hpp b/include/zxorm/orm/connection.hpp
index 9f00288..28da7c6 100644
--- a/include/zxorm/orm/connection.hpp
+++ b/include/zxorm/orm/connection.hpp
@@ -23,6 +23,9 @@
 #include <type_traits>
 #include <unordered_map>
 #include <utility>
+#include <cstddef>
+
+#define ssize_t std::ptrdiff_t
 
 namespace zxorm {
     template <class... Table>
diff --git a/include/zxorm/orm/expression.hpp b/include/zxorm/orm/expression.hpp
index 857a230..5660d98 100644
--- a/include/zxorm/orm/expression.hpp
+++ b/include/zxorm/orm/expression.hpp
@@ -20,7 +20,7 @@ namespace zxorm {
         NOT_LIKE,
         GLOB,
         NOT_GLOB,
-        IN,
+        _IN,
         NOT_IN,
     };
 
@@ -56,7 +56,7 @@ namespace zxorm {
             case comparison_op_t::NOT_GLOB:
                 out << " NOT GLOB ";
                 break;
-            case comparison_op_t::IN:
+            case comparison_op_t::_IN:
                 out << " IN ";
                 break;
             case comparison_op_t::NOT_IN:
@@ -140,7 +140,7 @@ namespace zxorm {
         std::string serialize() const {
             std::stringstream ss;
             ss << "`" << Table::name.value << "`.`" << Column::name.value << "` " << op;
-            if constexpr (op == comparison_op_t::IN || op == comparison_op_t::NOT_IN) {
+            if constexpr (op == comparison_op_t::_IN || op == comparison_op_t::NOT_IN) {
                 ss << " (";
                 for (size_t i = 0; i < to_bind.size(); i++) {
                     ss << "?,";
diff --git a/include/zxorm/orm/field.hpp b/include/zxorm/orm/field.hpp
index 8dd33ef..14d77d8 100644
--- a/include/zxorm/orm/field.hpp
+++ b/include/zxorm/orm/field.hpp
@@ -143,7 +143,7 @@ namespace zxorm {
 
         template <typename Container>
         requires(is_multi_value_binding<Container>::value)
-        ColumnExpression<Table, column_t, comparison_op_t::IN, Container> in(Container value) {
+        ColumnExpression<Table, column_t, comparison_op_t::_IN, Container> in(Container value) {
             return value;
         }
 
diff --git a/include/zxorm/orm/table.hpp b/include/zxorm/orm/table.hpp
index 37e8bc8..c4769bf 100644
--- a/include/zxorm/orm/table.hpp
+++ b/include/zxorm/orm/table.hpp
@@ -14,6 +14,7 @@
 #include "zxorm/orm/column.hpp"
 #include "zxorm/orm/field.hpp"
 #include "zxorm/error.hpp"
+#include "zxorm/orm/statement.hpp"
 
 namespace zxorm {
     namespace __foreign_key_detail {
